164796888 Password Reset Functionality

Simplemart17 opened the pull request:
#### Description
_As a User_ I want to be able to reset my forgotten password.
An authenticated users can reset their password.
#### Type of change
- [ ] Bug Fix (a non-breaking change which fixes an issue)
- [x] New Feature (a non-breaking change that adds functionality)
- [ ] New Chore (a non-breaking change that adds functionality)
- [ ] Breaking Change (a fix or feature which would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update
#### How Has This Been Tested?
- [x] End to End
- [x] Integration
#### Checklist:
- [x] My code follows the style guidelines of this project
- [x] New and existing unit tests pass locally with my changes
#### Flowchart
![flowchart](https://user-images.githubusercontent.com/41796727/56368911-476ecd80-61f0-11e9-914d-c2e663dca639.jpeg)
#### Screenshot
<img width="1440" alt="Screenshot 2019-04-18 at 3 41 33 PM" src="https://user-images.githubusercontent.com/41796727/56369049-82710100-61f0-11e9-9fee-c0ee1653cebd.png">

#### PT
[#164796888](https://www.pivotaltracker.com/story/show/164796888)

Simplemart17 committed:
164796888-feature(passwordReset): add function to reset password

- add password reset functionality
- modify the controller
- add integration test for password reset functionality

[Delivers #164796888]

chikeozulumba reviewed (commented):
@Simplemart17 Nice work done. LGTM

Temmyogunbo commented on src/controllers/authentication/passwordReset.js:
```diff
@@ -0,0 +1,48 @@
+    if (resetTokenExpires <= date) return res.status(401).json(errorResponseFormat({ status: 'fail', message: 'Token has expired!' }));
+
+    const { password } = req.body;
+    const hashPassword = bcrypt.hashSync(password, bcrypt.genSaltSync(10));
```
Let's refactor this and use the method from our user model.

Temmyogunbo commented on src/utils/auth/resetPassword.js:
```diff
@@ -0,0 +1,17 @@
+ * @param {object} body The request object
+ * @returns {function} Returns validation object
+ */
+export const validateResetField = async (body) => {
```
Why the async?

Temmyogunbo commented on tests/integration/passwordReset.test.js:
```diff
@@ -0,0 +1,82 @@
+    agent.put('/api/v1/password-reset/49876jkghdfvghj-dguye89063478-efdvuk789')
+      .send({
+        password: 'tyie56H#JJJ88',
+        password_confirmation: 'tyie56H#JJJ88'
```
Can we put this in a mock file?

Simplemart17 committed:
164796888-feature(passwordReset): refactor code

- refactor password reset function to make use of hash method from the users table

[Delivers #164796888]
